### PR TITLE
reworked mail_crypt guide to make things way more simple and prepare …

### PR DESCRIPTION
**Pull request by NorseGaud**
…for user folder encryption

# Description

Preparing for per folder/user encryption guide per https://github.com/docker-mailserver/docker-mailserver/issues/1904 discussion by cleaning up and simplifying the mail_crypt guide. Also, not enabling mail_crypt globally doesn't allow `doveadm mailbox cryptokey` to work: http://dovecot.2317879.n4.nabble.com/doveadm-mailbox-cryptokey-not-found-td58533.html

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change that does improve existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [x] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by wernerfred — DISMISSED** [review]
LGTM

**Review by georglauterbach — APPROVED** [review]

File: docs/content/config/security/mail_crypt.md [hunk 1]
```diff
@@ -10,22 +10,31 @@ title: 'Security | mail_crypt (email/storage encryption)'
...
+## Single Encryption Key / Global Method
 
-1. Before you can enable mail_crypt, you'll need to copy out several dovecot/conf.d files to the host (from a running container) and then take the container down:
-    ```bash
-    mkdir -p config/dovecot
-    docker cp mailserver:/etc/dovecot/conf.d/20-lmtp.conf config/dovecot/
-    docker cp mailserver:/etc/dovecot/conf.d/20-imap.conf config/dovecot/
-    docker cp mailserver:/etc/dovecot/conf.d/20-pop3.conf config/dovecot/
-    docker-compose down
+1. Create `10-custom.conf` and populate it with in the following:
```
Comment (casperklein) [inline docs/content/config/security/mail_crypt.md:21]:
```suggestion
1. Create `10-custom.conf` and populate it with the following:
```

File: docs/content/config/security/mail_crypt.md [hunk 2]
```diff
@@ -10,22 +10,31 @@ title: 'Security | mail_crypt (email/storage encryption)'
...
+    plugin {
+      mail_crypt_global_private_key = </certs/ecprivkey.pem
+      mail_crypt_global_public_key = </certs/ecpubkey.pem
+      mail_crypt_save_version = 2
+    }
     ```
-2. You then need to [generate your global EC key](https://doc.dovecot.org/configuration_manual/mail_crypt_plugin/#ec-key).
-3. The EC key needs to be available in the container. I prefer to mount a /certs directory into the container: 
+
+2. Shut down your mailserver (`docker-compose down`/etc)
```
Comment (casperklein) [inline docs/content/config/security/mail_crypt.md:33]:
```suggestion
2. Shutdown your mailserver (`docker-compose down`)
```

File: docs/content/config/security/mail_crypt.md [hunk 3]
```diff
@@ -10,22 +10,31 @@ title: 'Security | mail_crypt (email/storage encryption)'
...
+
+3. You then need to [generate your global EC key](https://doc.dovecot.org/configuration_manual/mail_crypt_plugin/#ec-key). We named them `/certs/ecprivkey.pem` and `/certs/ecpubkey.pem` in Step #1.
```
Comment (casperklein) [inline docs/content/config/security/mail_crypt.md:35]:
```suggestion
3. You then need to [generate your global EC key](https://doc.dovecot.org/configuration_manual/mail_crypt_plugin/#ec-key). We named them `/certs/ecprivkey.pem` and `/certs/ecpubkey.pem` in step #1.
```

File: docs/content/config/security/mail_crypt.md [hunk 4]
```diff
@@ -35,43 +44,24 @@ Official Dovecot documentation: https://doc.dovecot.org/configuration_manual/mai
           - ./certs/:/certs
         . . .
     ```
-4. While you're editing the docker-compose.yml, add the configuration files you copied out:
+
+5. While you're editing the docker-compose.yml, add the configuration file:
```
Comment (casperklein) [inline docs/content/config/security/mail_crypt.md:48]:
```suggestion
5. While you're editing the `docker-compose.yml`, add the configuration file:
```

File: docs/content/config/security/mail_crypt.md [hunk 5]
```diff
@@ -35,43 +44,24 @@ Official Dovecot documentation: https://doc.dovecot.org/configuration_manual/mai
...
-        }
-      }
-      ```
-    - If you use pop3, make the same changes in `20-pop3.conf`
-6. Start the container and monitor the logs for any errors
-
-This should be the minimum required for encryption of the mail while in storage.
\ No newline at end of file
+
+6. Start the container, monitor the logs for any errors, send yourself a message, and then confirm the file on disk is encrypted:
```
Comment (casperklein) [inline docs/content/config/security/mail_crypt.md:60]:
```suggestion
6. Start the container, monitor the logs for any errors, send yourself a message and then confirm the file on disk is encrypted:
```

Not sure if a comma is needed.

**Review by casperklein — CHANGES_REQUESTED** [review]

File: docs/content/config/security/mail_crypt.md (hunk 5 above)
Comment (NorseGaud) [reply docs/content/config/security/mail_crypt.md:60]: As written, the comma is necessary before the and.

**Review by casperklein — APPROVED** [review]

**Review by wernerfred — APPROVED** [review]

**Review by wernerfred — APPROVED** [review]

**Review by casperklein — CHANGES_REQUESTED** [review]

File: docs/content/config/security/mail_crypt.md [hunk 6]
```diff
@@ -35,43 +44,24 @@ Official Dovecot documentation: https://doc.dovecot.org/configuration_manual/mai
...
+    ```
+    [root@ip-XXXXXXXXXX ~]# cat /mnt/efs-us-west-2/maildata/awesomesite.com/me/cur/1623989305.M6v�z�@�� m}��,��9����B*�247.us-west-2.compute.inE��\Ck*�@7795,W=7947:2,
```
[the remark this answers is not in the document]
Comment (NorseGaud) [reply docs/content/config/security/mail_crypt.md:62]: It's me too. I just expect it 🙂

[the remark this answers is not in the document]
Comment (casperklein) [reply docs/content/config/security/mail_crypt.md]: Is that part of the encryption?

[the remark this answers is not in the document]
Comment (NorseGaud) [reply docs/content/config/security/mail_crypt.md]: Yep

**Review by georglauterbach — APPROVED** [review]

File: docs/content/config/security/mail_crypt.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (casperklein) [reply docs/content/config/security/mail_crypt.md]: Ok, then it makes sense. What about my `cat -A` suggestion?

**Review by casperklein — APPROVED** [review]

**Comment by wernerfred** [comment]
@all-contributors add @NorseGaud for documentation

**Comment by wernerfred** [comment]
@all-contributors add @NorseGaud for documentation

**Comment by wernerfred** [comment]
@all-contributors add @NorseGaud for documentation

**Pull request merged**